### PR TITLE
feat: add device management test

### PR DESCRIPTION
**Pull request by meenal06**

### CODE REVIEW

**Comment by meenal06** [comment]
@kegsay please review!

**Comment by meenal06** [comment]
@kegsay please review

File: tests/apidoc_device_management_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+			JSON: []match.JSON{
+				match.JSONKeyEqual("device_id", deviceID),
+				match.JSONKeyEqual("display_name", "device display"),
+			},
+		})
+	})
+}
+
+func getDevice(t *testing.T, authedClient *client.CSAPI, deviceID string) *http.Response {
+	res := authedClient.MustDo(t, "GET", []string{"_matrix", "client", "r0", "devices", deviceID}, nil)
```
Comment (kegsay) [inline tests/apidoc_device_management_test.go:45]: Again, it doesn't make sense to have a proxy function here.

File: tests/apidoc_device_management_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+func TestDeviceManagement(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+	unauthedClient := deployment.Client(t, "hs1", "")
+	authedClient := deployment.RegisterUser(t, "hs1", "test_device_management_user", "superuser")
+
+	// sytest: GET /device/{deviceId}
+	t.Run("GET /device/{deviceId}", func(t *testing.T) {
+		deviceID := "login_device"
+		_ = unauthedClient.MustDo(t, "POST", []string{"_matrix", "client", "r0", "login"}, json.RawMessage(`{
```
Comment (kegsay) [inline tests/apidoc_device_management_test.go:23]: Again, prefer `WithJSONBody` so you don't need to do escaping on `deviceID`.

**Review by kegsay — CHANGES_REQUESTED** [review]

**Comment by meenal06** [comment]
@kegsay I have addressed the review comments, please do review

File: tests/apidoc_device_management_test.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+			"device_id":                   deviceIDSecond,
+			"initial_device_display_name": "device display",
+		})
+		_ = unauthedClient.MustDoFunc(t, "POST", []string{"_matrix", "client", "r0", "login"}, reqBody)
+
+		res := authedClient.MustDoFunc(t, "GET", []string{"_matrix", "client", "r0", "devices"})
+
+		foundDeviceId := 0
+
+		// Complement also logged the login device used to create the authedClient, hence the use of the following hack
```
Comment (meenal06) [inline tests/apidoc_device_management_test.go:76]: @kegsay Any other method to remove this hack over here 😕

File: tests/apidoc_device_management_test.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+	// sytest:	GET /devices
+	t.Run("GET /devices", func(t *testing.T) {
+		deviceID := "login_device"
+		deviceIDSecond := "login_device_2"
+		reqBody := client.WithJSONBody(t, map[string]interface{}{
+			"type": "m.login.password",
+			"identifier": map[string]interface{}{
+				"type": "m.id.user",
+				"user": "@test_device_management_user:hs1",
```
Comment (kegsay) [inline tests/apidoc_device_management_test.go:64]: Use `authedClient.UserID` please.

File: tests/apidoc_device_management_test.go (hunk 3 above)
Comment (kegsay) [reply tests/apidoc_device_management_test.go:76]: You can do something like:
```go
wantDeviceIDs := map[string]bool{
    deviceID: true,
    deviceIDSecond: true,
}
res := authedClient.MustDoFunc(t, "GET", []string{"_matrix", "client", "r0", "devices"})
must.MatchResponse(t, res, match.HTTPResponse{
    StatusCode: 200,
    JSON: []match.JSON{
        match.JSONArrayEach("devices", func(r gjson.Result) error {
            gotDeviceID := r.Get("device_id").Str
            if wantDeviceIDs[gotDeviceID] {
                delete(wantDeviceIDs, gotDeviceID)
                return nil
            }
            // return nil here if unexpected devices are okay
            return fmt.Errorf("unexpected device ID: %s", gotDeviceID)
        }),
    },
})
if len(wantDeviceIDs) != 0 {
    t.Errorf("/devices did not return the following expected devices: %v", wantDeviceIDs)
}
```

**Review by kegsay — APPROVED** [review]
Excellent PR otherwise, thanks!

**Comment by meenal06** [comment]
Fixed @kegsay , please review & merge 😄

File: tests/apidoc_device_management_test.go [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+	authedClient := deployment.RegisterUser(t, "hs1", "test_device_management_user", "superuser")
+
+	// sytest: GET /device/{deviceId}
+	t.Run("GET /device/{deviceId}", func(t *testing.T) {
+		deviceID := "login_device"
+		reqBody := client.WithJSONBody(t, map[string]interface{}{
+			"type": "m.login.password",
+			"identifier": map[string]interface{}{
+				"type": "m.id.user",
+				"user": "@test_device_management_user:hs1",
```
Comment (kegsay) [inline tests/apidoc_device_management_test.go:27]:
```suggestion
				"user": authedClient.UserID,
```

**Pull request merged**